From 0a256df8495ff86a9afcfe1edc571f34e2d81532 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 22 Nov 2014 22:27:10 +0000
Subject: [PATCH 6/6] Search for freeaddrinfo in SOCKET_LIBS

.. using system-specifc headers so that the declaration
includes the correct calling convention.

Reverts one of the two changes from 888bc8.
---
 configure.ac | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index d576265..c790c83 100644
--- a/configure.ac
+++ b/configure.ac
@@ -298,7 +298,6 @@ case "${host_os}" in
   *os2*)
     SYS=os2
     LDFLAGS="${LDFLAGS} -Zomf -Zbin-files -Zargs-wild -Zhigh-mem"
-    AC_LIBOBJ([freeaddrinfo])
     ;;
   *)
     SYS="${host_os}"
@@ -649,6 +648,16 @@ AC_LINK_IFELSE([
         inet_pton(AF_INET, "127.0.0.1", dst);
     ])],[AC_DEFINE([HAVE_INET_PTON],[1],[Define to 1 if you have inet_pton function])],[AC_LIBOBJ([inet_pton])])
 AC_CHECK_FUNCS([if_nameindex if_nametoindex])
+AC_LINK_IFELSE(
+    [AC_LANG_PROGRAM([#ifdef _WIN32
+                     #include <winsock2.h>
+                     #include <ws2tcpip.h>
+                     #else
+                     #include <sys/socket.h>
+                     #include <netdb.h>
+                     #endif], [
+        freeaddrinfo(0);
+    ])],[AC_DEFINE([HAVE_FREEADDRINFO],[1],[Define to 1 if you have freeaddrinfo function])],[AC_LIBOBJ([freeaddrinfo])])
 VLC_RESTORE_FLAGS
 
 AS_IF([test -n "$SOCKET_LIBS"], [
-- 
2.1.3

